globus_portal_framework/views/base.py

```diff
@@ -298,6 +301,47 @@ def detail(request: HttpRequest, index: str, subject: str) ->  django.http.HttpR
     template = gsearch.get_template(index, tvers)
     return render(request, template, gsearch.get_subject(index, subject,
                                                          request.user))
+@xframe_options_sameorigin
+def render_asset(request: HttpRequest, index: str) -> HttpResponse:
+    """Render an HTML page representing a specific asset, suitable for embedding in an iframe"""
+    # TODO: implement authentication, validate against advertised collections for index
+    # Two ways to call view to specify what we want:
+    #   ?url=  - the url of a public asset somewhere on the web
+    #   ?collection=&path= - the globus collection information needed to show an asset privately - eventual!!!
+    tvers = gsearch.get_template_path('detail-render-asset.html', index=index)
+    template = gsearch.get_template(index, tvers)
+
+    page_options = {
+        # Placeholder until we add authorization handling later. For now, lock the page to public-only behaviors.
+        'is_authenticated': False,  # request.user.is_authenticated
+    }
+
+    render_options = {
+        # JS will auto-detect unless this is provided
+        'url': None, # set below
+        'token': None, # Placeholder for future functionality
+        'mode': request.GET.get('render_mode'),
+    }
```

Comment:
This may cause problems with Django Caching, since the `token` may be leaked across views.